### PR TITLE
[WIP] Fix deprecation issues for PHP 8.1 (Part 2)

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Fix deprecation warnings for PHP 8.1. compatibility.

### Related Issues
1. magento/magento2#34567
2. magento/magento2#34778

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by karyna-tsymbal-atwix** [comment]
@magento run Semantic Version Checker, Static Tests

**Review by andrewbess — APPROVED** [review]
Hello @karyna-tsymbal-atwix 
Thank you for your contribution
It looks great for me ✔️

**Comment by magento-engcom-team** [comment]
Hi @andrewbess, thank you for the review.
[ENGCOM-9367](https://jira.corp.magento.com/browse/ENGCOM-9367) has been created to process this Pull Request
:eight_spoked_asterisk: @andrewbess, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: lib/internal/Magento/Framework/Url/Decoder.php [hunk 1]
```diff
@@ -24,14 +24,14 @@ public function __construct(
...
-     * base64_decode() for URLs decoding
+     * Base64 decode for URL
      *
      * @param    string $url
      * @return   string
      */
     public function decode($url)
     {
-        $url = base64_decode(strtr($url, '-_,', '+/='));
+        $url = $url !== null ? base64_decode(strtr($url, '-_,', '+/=')) : '';
```
Comment (xmav) [inline lib/internal/Magento/Framework/Url/Decoder.php]: I would prefer to fix that in
https://github.com/magento/magento2/blob/6bbf0aa7fcec36592caf28536d9ac47b220094ca/app/code/Magento/Widget/Controller/Adminhtml/Widget/Index.php#L53-L55

**Comment by andrewbess** [comment]
@magento run Semantic Version Checker, Static Tests

File: lib/internal/Magento/Framework/Url/Decoder.php (hunk 1 above)
Comment (andrewbess) [reply lib/internal/Magento/Framework/Url/Decoder.php]: Hello @xmav 
Thank you for your review.
We fixed it, please check

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
Closed due to some fixes have been delivered into 2.4-develop and some fixes have been delivered by internal team

**Pull request closed without merging**